src/test/java/core/basesyntax/service/impl/FruitMapperImplTest.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.FruitMapper;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FruitMapperImplTest {
+    private static final String TITLE = "type,fruit,quantity";
+    private static final String FRUIT_APPLE = "apple";
+    private static final String FRUIT_BANANA = "banana";
+    private static final String VALID_FIRST_ROW = "b,banana,20";
+    private static final String VALID_SECOND_ROW = "s,apple,100";
+    private static final String NOT_VALID_ROW = "b,banana,notanumber";
+    private FruitMapper fruitMapper;
+
+    @BeforeEach
+    void setUp() {
+        fruitMapper = new FruitMapperImpl();
+    }
+
+    @Test
+    void validData_convert_oK() {
+        List<String> inputData = new ArrayList<>();
+        inputData.add(TITLE);
+        inputData.add(VALID_FIRST_ROW);
+        inputData.add(VALID_SECOND_ROW);
+        int quantityForBanana = 20;
+        int quantityForApple = 100;
+        List<Fruit> expected = List.of(
+                new Fruit(FRUIT_BANANA, Operation.BALANCE, quantityForBanana),
+                new Fruit(FRUIT_APPLE, Operation.SUPPLY, quantityForApple)
+        );
+        List<Fruit> actual = fruitMapper.convert(inputData);
+        assertEquals(expected, actual, "Fruits should be correctly mapped from input data.");
```

Comment:
Just to expand your knowledge a bit, there is a method and various constructions around it like `assertThat(actual).using...().ignoring...().isEqualTo(expected)` that is very useful in a more complex comarisions, which provides you with a more flexible approaches to asserts, like ignoring collection order, using of recursive field by field comparison ignoring overrided equals or some fields during the comparison, or in case you are using some other classes and there is no equals. Mostly useful in the assertion of collections but also has many overloaded variants for different types

Reply:
Thank you I`ll explore this question